cmd/openqa-mon/tui.go

```diff
@@ -304,23 +331,30 @@ func (tui *TUI) Update() {
 		lines++
 	}
 	if tui.showHelp {
-		help := "?:Toggle help    r: Refresh    d:Toggle notifications    b:Toggle bell    +/-:Modify refresh time    p:Toggle pause"
+		help := "?:Toggle help  r:Refresh  d:Toggle notifications  b:Toggle bell  +/-:Modify refresh time  p:Toggle pause  <>:Page"
 		if len(tui.Model.HideStates) > 0 {
-			help += "    h:Toggle hide"
+			help += "  h:Toggle hide"
 		}
-		help += "    q:Quit"
+		help += "  q:Quit"
 		PrintLine(help, width)
 		lines++
 	}
-	fmt.Println()
-	lines++
-
 	offset := 0
-	maxHeight := height
+	maxHeight := height - 2
 	if tui.showStatus {
-		maxHeight -= 2
+		maxHeight -= 1
 	}
-	for _, job := range tui.Model.jobs {
+	startIdx := tui.currentPage * maxHeight
+	endIdx := min(startIdx+maxHeight, len(tui.Model.jobs))
+	tui.totalPages = len(tui.Model.jobs) / maxHeight
+	if len(tui.Model.jobs)%maxHeight > 0 {
+		tui.totalPages++
+	}
+	// safety check
+	if startIdx >= len(tui.Model.jobs) || endIdx > len(tui.Model.jobs) {
```

Comment:
Instead of returning, I would set `startIdx` and `endIdx` to the maximum values:

```suggestion
	startIdx = min(startIdx, len(tui.Model.jobs)-1)
	endIdx = min(endIds, len(tui.Model.jobs)-1)
```
Like this the safety check ensure that the program still displays the jobs, otherwise we have a safety check that will prevent the program from crashing, but not displaying anything useful.

Reply:
There is still a glitch, when you have a small terminal (let's say 10 lines) and you enlarge it, the page recalc logic doesn't trigger immediately. To have a full responsive application we'd need to intercept the terminal resize event and act accordingly, but it would be easier to just use a proper terminal library as stated above :)

Reply:
Can you please add a comment or even better file an issue for this?

Reply:
added https://github.com/os-autoinst/openqa-mon/issues/158

Reply:
I think the `if ()` statement can be omitted.

Reply:
Also, why not apply them directly in line 348 and 349? :thinking: 

```go
startIdx := min(tui.currentPage * pageHeight, len(tui.Model.jobs)-1)
endIdx := min(startIdx+pageHeight, len(tui.Model.jobs)-1)
```